docs/upgrade-to-v6.md

```diff
@@ -0,0 +1,507 @@
+
+### Removal of `buildRequest()` 
+
+In the previous version of the SDK the `buildRequest()` method call was neccessary when building and calling requests.
+Previously a call to get the current user would look like the following: 
+```java
+User me = graphClient.me().buildRequest().get();
+```
+The same call would be the following in v6:
+```java 
+User me = graphClient.me().get();
+```
+
+### CollectionPage Types Changed to CollectionResponse Types
+
+The `CollectionPage` suffix has been changed to `CollectionResponse` to better reflect the type of object that is returned from a request. 
+For example, a call to get a collection of users would return the following in v5:
+```java
+UserCollectionPage users = graphClient.users().buildRequest().get();
+```
+In v6 the same call would return:
+```java
+UserCollectionResponse users = graphClient.users().get();
+```
+
+## Properties Now Accessed via Getters and Setters
+
+In v5 properties were accessed directly via the property name. In v6 properties are accessed via getters and setters.
+For example, in v5 a user would access the `displayName` property of a `User` object as follows:
+```java
+User me = graphClient.me().buildRequest().get();
+String displayName = me.displayName;
+```
+In v6 the same call would look like the following:
+```java
+User me = graphClient.me().get();
+String displayName = me.getDisplayName();
+```
+
+### Querying Collections
+
+Querying collections has changed in v6 to more closely resemble the response from the API. 
+Previously a user would prepare to query a collection as follows:
+```java
+UserCollectionPage response = graphClient.users()
+                            .buildRequest()
+                            .get();
+List<User> usersList = response.getCurrentPage();                        
+```
+In v6 the same call would look like the following:
+```java
+UserCollectionResponse response = graphClient.users().get();
+List<User> usersList = response.getValue();
+```
+
+### Change in Indexing via Improved Fluent API Pattern 
+
+The fluent API pattern has changed slightly in v6. Previously the fluent API pattern would index into a collection through an overload method call in the request builder pattern. In v6 we have added the `byId` suffix to make it obvious when a user is indexing into a collection.
+For example, retrieving a message by id would look like the following in v5:
+```java
+Message singleMessage = graphClient.me().messages("<Message Id>").buildRequest().get();
+``` 
+In v6 the same call would look like the following:
+```java
+Message singleMessage = graphClient.me().messages().byMessageId("<Message Id>").get();
+```
+
+### Option Class Removal
+
+The `Option` class has been removed and is no longer used to define query parameters, headers, or function parameters. These classes have been replaced with more specific and intuitive implementations. 
+
+### Headers 
```

Comment:
Maybe the `Headers` and the `Query Parameter` sections be subsections under `Option Class Removal`?